Stuck video at 99.88% when burn the subtitle SSA format

### Problem Description

hello,
im trying encode video to mp4 with "Very Fast 480p" + burn subtitle SSA format. it not finish in 3 hours (still continue) and stuck at 99.88%. you can try it by yourself.
download original file (demo).
https://anonfiles.com/o6i4a8Gby2/4335735ac0141cf007c211a6bcea22c9_mkv
https://gofile.io/d/J0ukIN

### Activity Log, Crash Log or any other details

```text
HandBrake 1.5.1 (2022011000)
OS: Microsoft Windows NT 10.0.22000.0
CPU: AMD Ryzen 5 1600 Six-Core Processor            
Ram: 16308 MB, 
GPU Information:
  NVIDIA GeForce GTX 1060 6GB - 31.0.15.2225
Screen: 1920x1080
Temp Dir: C:\Users\admin\AppData\Local\Temp\
Install Dir: C:\Program Files\HandBrake\
Data Dir: C:\Users\admin\AppData\Roaming\HandBrake

-------------------------------------------

 # Starting Encode ...

[22:11:51] base preset: Very Fast 480p30 (Modified)
[22:11:51] Remote Process started with Process ID: 16804 using port: 8037. Max Allowed Instances: 1
[22:11:51] Worker: Starting HandBrake Engine ...
[22:11:51] Worker: Starting Web Server on port 8037 ...
[22:11:52] Worker: Disconnected worker monitoring enabled!
[22:11:52] Compile-time hardening features are enabled
 libhb thread
[22:11:52] Starting work at: Sat Nov 05 22:11:52 2022
[22:11:52] 1 job(s) to process
[22:11:52] json job:
{
  "Audio": {
    "AudioList": [
      {
        "Bitrate": 96,
        "DRC": 0,
        "Encoder": "av_aac",
        "Gain": 0,
        "Mixdown": 4,
        "NormalizeMixLevel": false,
        "Samplerate": 0,
        "Name": "Japanese Stereo 2ch AAC",
        "Track": 0,
        "DitherMethod": 0
      }
    ],
    "CopyMask": [
      "copy:aac",
      "copy:ac3",
      "copy:dtshd",
      "copy:dts",
      "copy:eac3",
      "copy:flac",
      "copy:mp3",
      "copy:truehd",
      "copy:mp2"
    ],
    "FallbackEncoder": "ac3"
  },
  "Destination": {
    "ChapterList": [
      {
        "Name": "Opening Credits"
      },
      {
        "Name": "Part A"
      },
      {
        "Name": "Part B"
      },
      {
        "Name": "Part C"
      },
      {
        "Name": "Part D"
      },
      {
        "Name": "Preview"
      }
    ],
    "ChapterMarkers": true,
    "AlignAVStart": true,
    "File": "C:\\Users\\admin\\Videos\\4335735Ac0141cf007c211a6bcea22c9.m4v",
    "Mp4Options": {
      "IpodAtom": false,
      "Mp4Optimize": true
    },
    "Mux": "av_mp4"
  },
  "Filters": {
    "FilterList": [
      {
        "ID": 4,
        "Settings": {
          "mode": "7"
        }
      },
      {
        "ID": 3,
        "Settings": {
          "block-height": "16",
          "block-thresh": "80",
          "block-width": "16",
          "filter-mode": "1",
          "mode": "0",
          "motion-thresh": "2",
          "spatial-metric": "2",
          "spatial-thresh": "3"
        }
      },
      {
        "ID": 13,
        "Settings": {
          "crop-bottom": "0",
          "crop-left": "0",
          "crop-right": "0",
          "crop-top": "0",
          "height": "480",
          "width": "720"
        }
      },
      {
        "ID": 6,
        "Settings": {
          "mode": "2",
          "rate": "27000000/900000"
        }
      }
    ]
  },
  "PAR": {
    "Num": 32,
    "Den": 27
  },
  "Metadata": {},
  "SequenceID": 0,
  "Source": {
    "Angle": 1,
    "Range": {
      "Type": "chapter",
      "Start": 1,
      "End": 6
    },
    "Title": 1,
    "Path": "C:\\Users\\admin\\Desktop\\4335735ac0141cf007c211a6bcea22c9.mkv"
  },
  "Subtitle": {
    "Search": {
      "Burn": false,
      "Default": false,
      "Enable": false,
      "Forced": false
    },
    "SubtitleList": [
      {
        "Burn": true,
        "Default": false,
        "Forced": false,
        "ID": 1,
        "Offset": 0,
        "Track": 0,
        "Name": "English Subtitles"
      }
    ]
  },
  "Video": {
    "Encoder": "x264",
    "Level": "3.1",
    "TwoPass": false,
    "Turbo": false,
    "ColorMatrixCode": 0,
    "Options": "",
    "Preset": "veryfast",
    "Profile": "main",
    "Quality": 25,
    "QSV": {
      "Decode": false
    }
  }
}
[22:11:52] CPU:
[22:11:52]  - logical processor count: 12
[22:11:52] Intel Quick Sync Video support: no
[22:11:52] hb_scan: path=C:\Users\admin\Desktop\4335735ac0141cf007c211a6bcea22c9.mkv, title_index=1
udfread ERROR: ECMA 167 Volume Recognition failed
src/libbluray/disc/disc.c:333: failed opening UDF image C:\Users\admin\Desktop\4335735ac0141cf007c211a6bcea22c9.mkv
src/libbluray/disc/disc.c:437: error opening file BDMV\index.bdmv
src/libbluray/disc/disc.c:437: error opening file BDMV\BACKUP\index.bdmv
src/libbluray/bluray.c:2646: nav_get_title_list(C:\Users\admin\Desktop\4335735ac0141cf007c211a6bcea22c9.mkv\) failed
[22:11:52] bd: not a bd - trying as a stream/file instead
libdvdread: Encrypted DVD support unavailable.
lilibdvdread: DVDOpenFileUDF:UDFFindFile /VIDEO_TS/VIDEO_TS.IFO failed
bdvdread: Can't open file VIDEO_TS.IFO.
libdvdnav: vm: vm: failed to read VIDEO_TS.IFO
[22:11:52] dvd: not a dvd - trying as a stream/file instead
Input #0, matroska,webm, from 'C:\Users\admin\Desktop\4335735ac0141cf007c211a6bcea22c9.mkv':
  Metadata:
    encoder         : libebml v1.3.1 + libmatroska v1.4.2
    creation_time   : 2016-10-10T04:22:39.000000Z
  Duration: 00:22:50.01, start: 0.000000, bitrate: 2962 kb/s
  Chapters:
    Chapter #0:0: start 0.000000, end 80.046000
      Metadata:
        title           : Opening Credits
    Chapter #0:1: start 80.046000, end 661.368000
      Metadata:
        title           : Part A
    Chapter #0:2: start 661.368000, end 917.083000
      Metadata:
        title           : Part B
    Chapter #0:3: start 917.083000, end 1173.130000
      Metadata:
        title           : Part C
    Chapter #0:4: start 1173.130000, end 1355.063000
      Metadata:
        title           : Part D
    Chapter #0:5: start 1355.063000, end 1370.014000
      Metadata:
        title           : Preview
  Stream #0:0(jpn): Video: h264 (High), yuv420p(progressive), 1280x720 [SAR 1:1 DAR 16:9], 23.98 fps, 23.98 tbr, 1k tbn, 47.95 tbc (default)
    Metadata:
      title           : Episode 27 - Desperation! The Resolute Infiltration!
      BPS             : 2749997
      BPS-eng         : 2749997
      DURATION        : 00:22:49.994000000
      DURATION-eng    : 00:22:49.994000000
      NUMBER_OF_FRAMES: 32847
      NUMBER_OF_FRAMES-eng: 32847
      NUMBER_OF_BYTES : 470935039
      NUMBER_OF_BYTES-eng: 470935039
      _STATISTICS_WRITING_APP: mkvmerge v7.9.0 ('Birds') 64bit
      _STATISTICS_WRITING_APP-eng: mkvmerge v7.9.0 ('Birds') 64bit
      _STATISTICS_WRITING_DATE_UTC: 2016-10-10 04:22:39
      _STATISTICS_WRITING_DATE_UTC-eng: 2016-10-10 04:22:39
      _STATISTICS_TAGS: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES
      _STATISTICS_TAGS-eng: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES
  Stream #0:1(jpn): Audio: aac (LC), 48000 Hz, stereo, fltp (default)
    Metadata:
      title           : Japanese Stereo 2ch AAC
      _STATISTICS_WRITING_APP: mkvmerge v7.9.0 ('Birds') 64bit
      _STATISTICS_WRITING_APP-eng: mkvmerge v7.9.0 ('Birds') 64bit
      _STATISTICS_WRITING_DATE_UTC: 2016-10-10 04:22:39
      _STATISTICS_WRITING_DATE_UTC-eng: 2016-10-10 04:22:39
      _STATISTICS_TAGS: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES
      _STATISTICS_TAGS-eng: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES
      BPS             : 167079
      BPS-eng         : 167079
      DURATION        : 00:22:50.005000000
      DURATION-eng    : 00:22:50.005000000
      NUMBER_OF_FRAMES: 64219
      NUMBER_OF_FRAMES-eng: 64219
      NUMBER_OF_BYTES : 28612492
      NUMBER_OF_BYTES-eng: 28612492
  Stream #0:2(eng): Subtitle: ass (default)
    Metadata:
      title           : English Subtitles
      _STATISTICS_WRITING_APP: mkvmerge v7.9.0 ('Birds') 64bit
      _STATISTICS_WRITING_APP-eng: mkvmerge v7.9.0 ('Birds') 64bit
      _STATISTICS_WRITING_DATE_UTC: 2016-10-10 04:22:39
      _STATISTICS_WRITING_DATE_UTC-eng: 2016-10-10 04:22:39
      _STATISTICS_TAGS: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES
      _STATISTICS_TAGS-eng: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES
      BPS             : 25862
      BPS-eng         : 25862
      DURATION        : 00:22:43.530000000
      DURATION-eng    : 00:22:43.530000000
      NUMBER_OF_FRAMES: 27603
      NUMBER_OF_FRAMES-eng: 27603
      NUMBER_OF_BYTES : 4408085
      NUMBER_OF_BYTES-eng: 4408085
  Stream #0:3: Attachment: ttf
    Metadata:
      filename        : Anderson Fireball XL5.ttf
      mimetype        : application/x-truetype-font
  Stream #0:4: Attachment: ttf
    Metadata:
      filename        : Bolton.ttf
      mimetype        : application/x-truetype-font
  Stream #0:5: Attachment: ttf
    Metadata:
      filename        : BoltonBold.ttf
      mimetype        : application/x-truetype-font
  Stream #0:6: Attachment: ttf
    Metadata:
      filename        : calibrib.ttf
      mimetype        : application/x-truetype-font
  Stream #0:7: Attachment: ttf
    Metadata:
      filename        : candarab.ttf
      mimetype        : application/x-truetype-font
  Stream #0:8: Attachment: ttf
    Metadata:
      filename        : DeBorstel Brush-Reduced.ttf
      mimetype        : application/x-truetype-font
  Stream #0:9: Attachment: otf
    Metadata:
      filename        : DK Ruby Red.otf
      mimetype        : application/vnd.ms-opentype
  Stream #0:10: Attachment: ttf
    Metadata:
      filename        : JELLYBELLY.TTF
      mimetype        : application/x-truetype-font
  Stream #0:11: Attachment: ttf
    Metadata:
      filename        : Matias Font.ttf
      mimetype        : application/x-truetype-font
  Stream #0:12: Attachment: ttf
    Metadata:
      filename        : nrkis.ttf
      mimetype        : application/x-truetype-font
  Stream #0:13: Attachment: ttf
    Metadata:
      filename        : RainyDays.ttf
      mimetype        : application/x-truetype-font
  Stream #0:14: Attachment: ttf
    Metadata:
      filename        : Rambla-Bold.ttf
      mimetype        : application/x-truetype-font
  Stream #0:15: Attachment: ttf
    Metadata:
      filename        : Rambla-BoldItalic.ttf
      mimetype        : application/x-truetype-font
  Stream #0:16: Attachment: ttf
    Metadata:
      filename        : Rambla-Italic.ttf
      mimetype        : application/x-truetype-font
  Stream #0:17: Attachment: ttf
    Metadata:
      filename        : Rambla-Regular.ttf
      mimetype        : application/x-truetype-font
  Stream #0:18: Attachment: ttf
    Metadata:
      filename        : Roboto-Black.ttf
      mimetype        : application/x-truetype-font
[22:11:52] scan: decoding previews for title 1
[22:11:52] scan: audio 0x1: aac, rate=48000Hz, bitrate=1 ??? (AAC LC) (2.0 ch)
[22:11:52] scan: 10 previews, 1280x720, 23.976 fps, autocrop = 0/0/0/0, aspect 16:9, PAR 1:1, color profile: 1-1-1, chroma location: left
[22:11:52] libhb: scan thread found 1 valid title(s)
[22:11:52] Starting Task: Encoding Pass
[22:11:52] [ass] libass API version: 0x1502000
[22:11:52] [ass] libass source: tarball: 0.15.2
[22:11:52] [ass] Shaper: FriBidi 1.0.11 (SIMPLE) HarfBuzz-ng 3.1.2 (COMPLEX)
[22:11:52] [ass] Using font provider directwrite (with GDI)
[22:11:52] work: track 1, dithering not supported by codec
[22:11:52] job configuration:
[22:11:52]  * source
[22:11:52]    + C:\Users\admin\Desktop\4335735ac0141cf007c211a6bcea22c9.mkv
[22:11:52]    + title 1, chapter(s) 1 to 6
[22:11:52]    + container: matroska,webm
[22:11:52]    + data rate: 2962 kbps
[22:11:52]  * destination
[22:11:52]    + C:\Users\admin\Videos\4335735Ac0141cf007c211a6bcea22c9.m4v
[22:11:52]    + container: MPEG-4 (libavformat)
[22:11:52]      + optimized for HTTP streaming (fast start)
[22:11:52]      + align initial A/V stream timestamps
[22:11:52]      + chapter markers
[22:11:52]  * video track
[22:11:52]    + decoder: h264 8-bit (yuv420p)
[22:11:52]    + filters
[22:11:52]      + Comb Detect (mode=0:spatial-metric=2:motion-thresh=2:spatial-thresh=3:filter-mode=1:block-thresh=80:block-width=16:block-height=16)
[22:11:52]      + Decomb (mode=39)
[22:11:52]      + Framerate Shaper (mode=2:rate=27000000/900000)
[22:11:52]        + frame rate: 23.976 fps -> peak rate limited to 30.000 fps
[22:11:52]      + Subtitle renderer ()
[22:11:52]      + Crop and Scale (width=720:height=480:crop-top=0:crop-bottom=0:crop-left=0:crop-right=0)
[22:11:52]        + source: 1280 * 720, crop (0/0/0/0): 1280 * 720, scale: 720 * 480
[22:11:52]    + Output geometry
[22:11:52]      + storage dimensions: 720 x 480
[22:11:52]      + pixel aspect ratio: 32 : 27
[22:11:52]      + display dimensions: 853 x 480
[22:11:52]    + encoder: H.264 (libx264)
[22:11:52]      + preset:  veryfast
[22:11:52]      + profile: main
[22:11:52]      + level:   3.1
[22:11:52]      + quality: 25.00 (RF)
[22:11:52]      + color profile: 1-1-1
[22:11:52]      + chroma location: left
[22:11:52]  * subtitle track 1, English [SSA] (track 0, id 0x2, Text) -> Render/Burn-in
[22:11:52]    + name: English Subtitles
[22:11:52]  * audio track 1
[22:11:52]    + name: Japanese Stereo 2ch AAC
[22:11:52]    + decoder: ??? (AAC LC) (2.0 ch) (track 1, id 0x1)
[22:11:52]      + samplerate: 48000 Hz
[22:11:52]    + mixdown: Stereo
[22:11:52]    + encoder: AAC (libavcodec)
[22:11:52]      + bitrate: 96 kbps, samplerate: 48000 Hz
[22:11:52] sync: expecting 32847 video frames
[22:11:52] encx264: min-keyint: 24, keyint: 240
[22:11:52] encx264: encoding at constant RF 25.000000
[22:11:52] encx264: unparsed options: level=3.1:ref=1:8x8dct=0:weightp=1:subme=2:mixed-refs=0:trellis=0:vbv-bufsize=14000:vbv-maxrate=14000:rc-lookahead=10
x264 [info]: using SAR=32/27
x264 [info]: using cpu capabilities: MMX2 SSE2Fast SSSE3 SSE4.2 AVX FMA3 BMI2 AVX2
x264 [info]: profile Main, level 3.1, 4:2:0, 8-bit
[22:11:52] sync: first pts video is 0
[22:11:52] sync: "Opening Credits" (1) at frame 1 time 0
[22:11:52] sync: first pts audio 0x1 is 0
[22:11:53] sync: first pts subtitle 0x2 is 576000
[22:11:53] [ass] fontselect: (DK Ruby Red, 400, 0) -> DKRubyRed, 0, DKRubyRed
[22:11:53] [ass] fontselect: (Calibri, 700, 0) -> Calibri-Bold, 0, Calibri-Bold
[22:12:02] sync: "Part A" (2) at frame 1930 time 7240983
[22:12:02] [ass] fontselect: (Rambla, 700, 0) -> Rambla-Bold, 0, Rambla-Bold
[22:12:02] [ass] fontselect: (Rambla, 700, 100) -> Rambla-BoldItalic, 0, Rambla-BoldItalic
[22:13:04] sync: "Part B" (3) at frame 15862 time 59538228
[22:13:32] sync: "Part C" (4) at frame 21994 time 82556223
[22:14:00] sync: "Part D" (5) at frame 28129 time 105585480
[22:14:18] sync: "Preview" (6) at frame 32491 time 121959337
[22:14:18] [ass] fontselect: (BoltonBold, 400, 0) -> BoltonBold, 0, BoltonBold
[22:14:20] reader: done. 1 scr changes
```


### What Operating System are you running?

Windows 11 Pro for Workstations (21H2)

### What version of HandBrake are you running?

HandBrake 1.5.1 (2022011000)

### Where did you download HandBrake from?

handbrake.fr

--- a/libhb/work.c
+++ b/libhb/work.c
@@ -1657,12 +1657,12 @@ static void do_job(hb_job_t *job)
         if (subtitle->source != IMPORTSRT &&
             subtitle->source != IMPORTSSA)
         {
-            subtitle->fifo_in  = hb_fifo_init( FIFO_SMALL, FIFO_SMALL_WAKE );
+            subtitle->fifo_in  = hb_fifo_init( FIFO_UNBOUNDED, FIFO_UNBOUNDED_WAKE );
         }
         if (!job->indepth_scan)
         {
             // When doing subtitle indepth scan, the pipeline ends at sync
-            subtitle->fifo_out = hb_fifo_init( FIFO_SMALL, FIFO_SMALL_WAKE );
+            subtitle->fifo_out = hb_fifo_init( FIFO_UNBOUNDED, FIFO_UNBOUNDED_WAKE );
         }
 
         w->fifo_in = subtitle->fifo_in;

